examples/pytorch/schnet/Alchemy_dataset.py

```diff
@@ -0,0 +1,224 @@
+from dgl.data.utils import download
+from dgl.data.utils import extract_archive
+import torch
+from torch.utils.data import Dataset
+from torch.utils.data import DataLoader
+from collections import namedtuple
+import pathlib
+import pandas as pd
+import numpy as np
+
+_urls = {'Alchemy': 'https://alchemy.tencent.com/data/'}
+
+
+class AlchemyBatcher:
+    def __init__(self, graph=None, label=None):
+        self.graph = graph
+        self.label = label
+
+
+def batcher():
+    def batcher_dev(batch):
+        graphs, labels = zip(*batch)
+        batch_graphs = dgl.batch(graphs)
+        labels = torch.stack(labels, 0)
+        return AlchemyBatcher(graph=batch_graphs, label=labels)
+
+    return batcher_dev
+
+
+class TencentAlchemyDataset(Dataset):
+
+    fdef_name = osp.join(RDConfig.RDDataDir, 'BaseFeatures.fdef')
+    chem_feature_factory = ChemicalFeatures.BuildFeatureFactory(fdef_name)
+
+    def alchemy_nodes(self, g):
+        feat = []
+        for n, d in g.nodes(data=True):
+            h_t = []
+            # Atom type (One-hot H, C, N, O F)
+            h_t += [
+                int(d['node_str'] == x)
+                for x in ['H', 'C', 'N', 'O', 'F', 'S', 'Cl']
+            ]
+            # Atomic number
+            h_t.append(d['node_type'])
+            # Acceptor
+            h_t.append(d['acceptor'])
+            # Donor
+            h_t.append(d['donor'])
+            # Aromatic
+            h_t.append(int(d['aromatic']))
+            # Hybradization
+            h_t += [
+                int(d['hybridization'] == x)
+                for x in (Chem.rdchem.HybridizationType.SP,
+                          Chem.rdchem.HybridizationType.SP2,
+                          Chem.rdchem.HybridizationType.SP3)
+            ]
+            h_t.append(d['num_h'])
+            feat.append((n, torch.FloatTensor(h_t)))
+
+        nx.set_node_attributes(g, dict(feat), "n_feat")
+
+    def alchemy_edges(self, g):
+        e = {}
+        for n1, n2, d in g.edges(data=True):
+            e_t = [
+                float(d['b_type'] == x)
+                for x in (Chem.rdchem.BondType.SINGLE,
+                          Chem.rdchem.BondType.DOUBLE,
+                          Chem.rdchem.BondType.TRIPLE,
+                          Chem.rdchem.BondType.AROMATIC, "NoBond")
+            ]
+            e[(n1, n2)] = e_t
+        nx.set_edge_attributes(g, e, "e_feat")
+
+    # sdf file reader for Alchemy dataset
+    def sdf_graph_reader(self, sdf_file):
+
+        with open(str(sdf_file), 'r') as f:
+            sdf_string = f.read()
+        mol = Chem.MolFromMolBlock(sdf_string, removeHs=False)
+        if mol is None:
+            print("rdkit can not parsing", sdf_file)
+            return None
+        feats = self.chem_feature_factory.GetFeaturesForMol(mol)
+
+        g = nx.DiGraph()
+        l = torch.FloatTensor(self.target.loc[int(sdf_file.stem)].tolist()) \
+            if self.mode == 'dev' else torch.LongTensor([int(sdf_file.stem)])
```

Comment:
Just curious, why do we have `FloatTensor` labels for the training set and `LongTensor` labels for the validation/test set? Maybe add some comments?